### PR TITLE
feat: improve reliability of connectivity module using time_sleep

### LINKED ISSUE #72: VNet Building Block flaky on first run due to azure eventually consistent permissions
```text
Plan: 5 to add, 0 to change, 0 to destroy.
azurerm_resource_group.spoke_rg: Creating...
azurerm_resource_group.spoke_rg: Creation complete after 1s [id=/subscriptions/b53caa89-ba38-4f04-a0b2-f989c98d4add/resourceGroups/connectivity]
azurerm_role_assignment.spoke_rg: Creating...
azurerm_role_assignment.spoke_rg: Still creating... [10s elapsed]
azurerm_role_assignment.spoke_rg: Still creating... [20s elapsed]
azurerm_role_assignment.spoke_rg: Creation complete after 25s [id=/subscriptions/b53caa89-ba38-4f04-a0b2-f989c98d4add/resourceGroups/connectivity/providers/Microsoft.Authorization/roleAssignments/64bf764d-6241-841f-0047-c4e9a73a0ce5]
azurerm_virtual_network.spoke_vnet: Creating...


Error: checking for presence of existing Virtual Network (Subscription: "b53caa89-ba38-4f04-a0b2-f989c98d4add"
Resource Group Name: "connectivity"
Virtual Network Name: "vnet-demo-app-vnet"): network.VirtualNetworksClient#Get: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="AuthorizationFailed" Message="The client 'dca119ef-092d-41ce-83d3-920d4eda271a' with object id 'dca119ef-092d-41ce-83d3-920d4eda271a' does not have authorization to perform action 'Microsoft.Network/virtualNetworks/read' over scope '/subscriptions/b53caa89-ba38-4f04-a0b2-f989c98d4add/resourceGroups/connectivity/providers/Microsoft.Network/virtualNetworks/vnet-demo-app-vnet' or the scope is invalid. If access was recently granted, please refresh your credentials."

  with azurerm_virtual_network.spoke_vnet,
  on main.tf line 29, in resource "azurerm_virtual_network" "spoke_vnet":
  29: resource "azurerm_virtual_network" "spoke_vnet" {
```

This is because the scope (the RG) was just created a few seconds before and Azure is eventually consistent. And this is even though the TF provider does wait until Azure returns the right data...

### PR DESCRIPTION
**Pull request by JohannesRudolph**
This is certainly not an elegant nor always successful workaround.
However it should significantly increase the "first time deploy"
reliability of this buildingblock.

closes #72

### CODE REVIEW

**Comment by JohannesRudolph** [comment]
- [ ]  when merging this to prod, also update the BBDef in meshStack to use this newer version

File: kit/azure/buildingblocks/connectivity/buildingblock/main.tf [hunk 1]
```diff
@@ -23,12 +28,23 @@ resource "azurerm_role_assignment" "spoke_rg" {
   scope                = azurerm_resource_group.spoke_rg.id
 }
 
+# Wait for the role assignment to become effective, azure is eventually consistent and thus sometimes flaky
+# even though the azurerm_role_assignment provider tries to cover for that, it's not always effective.
+# Example:
+# > Error: checking for presence of existing Virtual Network (Subscription: "b53caa89-ba38-4f04-a0b2-f989c98d4add"
+# > Resource Group Name: "connectivity"
+# > Virtual Network Name: "vnet-demo-app-vnet"): network.VirtualNetworksClient#Get: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="AuthorizationFailed" Message="The client 'dca119ef-092d-41ce-83d3-920d4eda271a' with object id 'dca119ef-092d-41ce-83d3-920d4eda271a' does not have authorization to perform action 'Microsoft.Network/virtualNetworks/read' over scope '/subscriptions/b53caa89-ba38-4f04-a0b2-f989c98d4add/resourceGroups/connectivity/providers/Microsoft.Network/virtualNetworks/vnet-demo-app-vnet' or the scope is invalid. If access was recently granted, please refresh your credentials."
```
Comment (florianow) [inline kit/azure/buildingblocks/connectivity/buildingblock/main.tf:36]: we should find a a other place for comments like that. For collie-hub its not relevant but would be part of the buildingblock

**Review by florianow — APPROVED** [review]
lgtm, only one comment

File: kit/azure/buildingblocks/connectivity/buildingblock/main.tf (hunk 1 above)
Comment (JohannesRudolph) [reply kit/azure/buildingblocks/connectivity/buildingblock/main.tf:36]: As discussed, this is a justification for the design decision to put the sleep in (which is usually an antipattern...)

**Comment by JohannesRudolph** [comment]
note: we did test this extensively and while it makes BB execution more reliable, it's not a deterministic solution. We have an idea how to solve that though...

**Pull request closed without merging**